RecoLocalMuon/GEMSegment/plugins/GEMSegmentBuilder.cc

```diff
@@ -11,15 +11,20 @@
 
 GEMSegmentBuilder::GEMSegmentBuilder(const edm::ParameterSet& ps) : geom_(nullptr) {
   // Algo name
-  algoName = ps.getParameter<std::string>("algo_name");
+  segAlgoName = ps.getParameter<std::string>("algo_name");
+  ge0AlgoName = ps.getParameter<std::string>("ge0_name");
 
-  edm::LogVerbatim("GEMSegmentBuilder") << "GEMSegmentBuilder algorithm name: " << algoName;
+  edm::LogVerbatim("GEMSegmentBuilder") << "GEMSegmentBuilder algorithm : ge0 name : " << ge0AlgoName
+                                        << " name: " << segAlgoName;
 
   // SegAlgo parameter set
   segAlgoPSet = ps.getParameter<edm::ParameterSet>("algo_pset");
+  ge0AlgoPSet = ps.getParameter<edm::ParameterSet>("ge0_pset");
 
   // Ask factory to build this algorithm, giving it appropriate ParameterSet
-  algo = GEMSegmentBuilderPluginFactory::get()->create(algoName, segAlgoPSet);
+  segAlgo = GEMSegmentBuilderPluginFactory::get()->create(segAlgoName, segAlgoPSet);
+  // Ask factory to build this algorithm, giving it appropriate ParameterSet
```

Comment:
Having twice the same comment is probably redundant...
If you remove one, then in the other one: "this algorithm, giving it" -> "these algorithms, giving them"

Reply:
Yep, updated